File: lib/console.ml
let error line col msg =
    Printf.printf "%s\n" line;
    Printf.printf "%*s" (col + 2) "^\n";
    Printf.printf "%*s" (col + 1) "|";
    Printf.printf "%s  " (String.make (String.length(line) - col + 5) '_');
    Printf.printf "%s %s\n" "\x1b[33;1merror:\x1b[0m" msg;

File: src/console.rs
//! Terminal diagnostics.

const YELLOW: &str = "\x1b[33;1m";
const RESET: &str = "\x1b[0m";

/// Print the offending source `line`, a caret under column `col`, and a yellow
/// `error:` message. This is a cleaned-up version of the OCaml `console.ml`,
/// whose `Printf` width arithmetic was fragile; the intent (line + caret +
/// message) is preserved.
pub fn error(line: &str, col: usize, msg: &str) {
    let pad = " ".repeat(col);
    eprintln!("{line}");
    eprintln!("{pad}^");
    eprintln!("{pad}{YELLOW}error:{RESET} {msg}");
}
